### PR TITLE
Add documentation for props passed to custom fields and widgets

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This adds some documentation for the custom fields and widget props that can be used

## Acceptance criteria
- [ ] Documentation makes sense to the design system team

## Definition of done
- [ ] Changes have been tested in vets-website
- [ ] Changes have been tested in IE11, if applicable
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.mdx
Comment (bkjohnson) [inline packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.mdx:57]: The `value` here contains the form data in the form of a single input.

File: packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.mdx
Comment (bkjohnson) [inline packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.mdx:63]: And here, `formData` presumably contains the entire form data. Is it worth calling this difference out a bit more, since the descriptions are the same? cc @cvalarida

**Review by bkjohnson — COMMENTED** [review]
Just a small question.

File: packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.mdx
Comment (jbalboni) [reply packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.mdx:63]: No, this is just the data for the particular field you're on, so it's the same as `value` for the widget. I'm not sure RJSF has named them differently, but you can see `StringField` pass `formData` directly to widgets as `value`: https://github.com/department-of-veterans-affairs/react-jsonschema-form/blob/master/src/components/fields/StringField.js#L40

**Review by bkjohnson — APPROVED** [review]

**Comment by jbalboni** [comment]
@bkjohnson @cvalarida I don't know how to get this merged, given the build failures

**Comment by bkjohnson** [comment]
@jbalboni the build checks aren't required, but this is being blocked by a review from @department-of-veterans-affairs/frontend-review-group

File: packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.mdx
Comment (billfienberg) [inline packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.mdx:59]: This sentence is confusing to me:
>The props they receive are derived the more broad props of field components.

**Review by billfienberg — APPROVED** [review]

**Pull request merged**
